packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -3,98 +3,169 @@ pragma solidity 0.8.15;
 
     /// @notice The address of the guardian, which can pause withdrawals from the System.
     ///         It can only be modified by an upgrade.
-    bytes32 public constant GUARDIAN_SLOT = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
+    address public guardian;
+
+    /// @notice Mapping of pause identifiers to their pause timestamps
+    mapping(address => uint256) public pauseTimestamps;
 
     /// @notice Emitted when the pause is triggered.
     /// @param identifier A string helping to identify provenance of the pause transaction.
     event Paused(string identifier);
 
     /// @notice Emitted when the pause is lifted.
-    event Unpaused();
+    event Unpaused(string identifier);
 
     /// @notice Emitted when configuration is updated.
     /// @param updateType Type of update.
     /// @param data       Encoded update data.
     event ConfigUpdate(UpdateType indexed updateType, bytes data);
 
     /// @notice Semantic version.
-    /// @custom:semver 1.2.0
-    string public constant version = "1.2.0";
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
 
     /// @notice Constructs the SuperchainConfig contract.
-    constructor() {
+    constructor() ReinitializableBase(2) {
         _disableInitializers();
     }
 
     /// @notice Initializer.
     /// @param _guardian    Address of the guardian, can pause the OptimismPortal.
-    /// @param _paused      Initial paused status.
-    function initialize(address _guardian, bool _paused) external initializer {
+    function initialize(address _guardian) external reinitializer(initVersion()) {
         _setGuardian(_guardian);
-        if (_paused) {
-            _pause("Initializer paused");
+    }
+
+    /// @notice Upgrades the SuperchainConfig contract.
+    function upgrade() external reinitializer(initVersion()) {
+        // Transfer the guardian into the new variable and clear the old storage slot.
+        bytes32 guardianSlot = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
+        _setGuardian(Storage.getAddress(guardianSlot));
+        Storage.setBytes32(guardianSlot, bytes32(0));
+
+        // Clear the old paused slot.
+        // Note that if the pause was active while the upgrade was happening, the system will no
+        // longer be paused after the upgrade. Upgrades should generally not ever be executed while
+        // the system is paused, but it's worth noting that this is the case.
+        bytes32 pausedSlot = bytes32(uint256(keccak256("superchainConfig.paused")) - 1);
+        Storage.setBytes32(pausedSlot, bytes32(0));
+    }
+
+    /// @notice Returns the duration after which a pause expires.
+    /// @return The duration after which a pause expires.
+    function pauseExpiry() external pure returns (uint256) {
+        return PAUSE_EXPIRY;
+    }
+
+    /// @notice Pauses the system for a specific identifier.
+    /// @param _identifier The address identifier for the pause.
```

Comment:
I realize that the name of the argument was `_identifier` and so using that name keeps the diff down, but is it the ideal name here? 
It feels vague to me.

Reply:
Identifier is okay, I think. We refer to it as the "Pause Identifier" in the spec.

Reply:
But we could improve the comment, I agree

Reply:
This comes from the design doc & spec and was already agreed on, I think it makes sense but open to changing it of someone else supports this